src/stdlib_logger.f90

```diff
@@ -1071,36 +1073,38 @@ subroutine log_message( self, message, module, procedure, prefix )
         call format_output_string( self,                         &
                                    d_and_t // m_and_p // pref // &
                                    trim( message ),              &
-                                   '    ' )
+                                   '    ',                       &
+                                   len_buffer,                   &
+                                   buffer)
 
         if ( self % units == 0 ) then
             if ( self % add_blank_line ) then
                 write( output_unit, '(a)', err=999, iostat=iostat, &
                         iomsg=iomsg) &
-                    new_line('a') // self % buffer(1:self % len_buffer)
+                    new_line('a') // buffer(1:len_buffer)
             else
                 write( output_unit, '(a)', err=999, iostat=iostat, &
                         iomsg=iomsg ) &
-                    self % buffer(1:self % len_buffer)
+                    buffer(1:len_buffer)
             end if
         else
             if ( self % add_blank_line ) then
                 do unit=1, self % units
                     write( self % log_units(unit), '(a)', err=999, iostat=iostat, &
                         iomsg=iomsg ) new_line('a') // &
-                        self % buffer(1:self % len_buffer)
+                        buffer(1:len_buffer)
                 end do
             else
                 do unit=1, self % units
                     write( self % log_units(unit), '(a)', err=999, iostat=iostat, &
                         iomsg=iomsg ) &
-                        self % buffer(1:self % len_buffer)
+                        buffer(1:len_buffer)
                 end do
             end if
         end if
 
-        deallocate( self % buffer )
-        self % len_buffer = 0
+        deallocate( buffer )
+        len_buffer = 0
```

Comment:
Deallocate and reinitialization not necessary here.
```suggestion
```

Reply:
True. Thank you for spotting it.